framework/packages/abstract-client/src/client.rs

```diff
@@ -97,6 +120,94 @@ mod test_utils {
         }
     }
 
+    pub mod cw20_builder {
+        // Re-exports to limit dependencies for consumer.
+        pub use cw20::{msg::Cw20ExecuteMsgFns, *};
+        pub use cw20_base::msg::{InstantiateMarketingInfo, QueryMsgFns as Cw20QueryMsgFns};
+        pub use cw_plus_interface::cw20_base::Cw20Base;
+
+        use cosmwasm_std::Addr;
+
+        use cw_orch::prelude::{CwOrchInstantiate, CwOrchUpload, Mock};
+        use cw_plus_interface::cw20_base::InstantiateMsg;
+
+        use crate::client::AbstractClientResult;
+
+        pub struct Cw20Builder {
+            chain: Mock,
```

Comment:
```suggestion
        pub struct Cw20Builder<Chain: CwEnv> {
            chain: Chain,
```
Can't it take any CwEnv chain?

Reply:
True, not 100% if we want to expose this in a non-test environment though.

Reply:
Sure, I don't mind making it available at the `CwEnv` level if there is consensus, the one thing I am not sure about is usually chains already have a cw20 code-id present right? Would it be a unusual to `upload` a new one as we do here?

Reply:
It could be. It could use the abstract version control to check if it's registered and use that code-id. But i'm not sure that's something that makes sense to include here...